### PR TITLE
Envest/download tcga data freeze

### PR DESCRIPTION
**Pull request by envest**
This PR creates a new script (`download_TCGA_data.sh`) to download TCGA data. Some of that data is from the TCGA pan-can atlas data freeze, and some is from refine.bio.

Over the course of our project discussions, we have elected to use the previously existing BRCA array and RNA-seq expression data. I have left code in the script to allow for a different decision in the future. For now, it is commented out with explanation.

Steps in the script:

1. Obtain TCGA data freeze manifest file
2. Download TCGA mutations, CNV, clinical, and RNA-seq data
3. Download GBM and AML array expression data with refine.bio client
4. #COMMENTED OUT Download BRCA array expression data from TCGA legacy archive

To run: `bash download_TCGA_data.sh` from any location. Downloaded manifest files go in the `data/` as well as downloaded data files (~104GB). The docker image is `envest/rnaseq_titration_results`.

A high level question I have is: I manually curated the BRCA legacy archive manifest, as in I selected search parameters by hand and downloaded the manifest file from https://portal.gdc.cancer.gov/legacy-archive. I don't know how to generate that manifest using a script. For now, we are not using that manifest at all. Should I keep that file around as if we were using it, if so, where? Right now my idea is to keep it in the `data/` folder, but previously that had been kept clean (.empty) upon first cloning. Related, in another simmering PR I add the gdc-client to the docker image. If we aren't using it _now_, should I add it? Commented out inside Dockerfile? Thank you :)

### CODE REVIEW

**Comment by envest** [comment]
Hmmm thinking maybe we should integrate the existing BRCA data download with the new download script... one script to download them all

**Comment by jaclyn-taroni** [comment]
I haven't had the chance to take a look yet but I agree with the principle of

> one script to download them all

certainly!

**Comment by jaclyn-taroni** [comment]
To respond to your questions:

> For now, we are not using that manifest at all. Should I keep that file around as if we were using it, if so, where? Right now my idea is to keep it in the data/ folder, but previously that had been kept clean (.empty) upon first cloning.

In this particular project structure (which I am not here to defend necessarily!), it seems like you might expect to find that in the root of the repository where `brca_data_urls.txt` is. Based on `.gitignore`, I am assuming you had do some `-force`-ing to get that in its current spot (`data/gdc_legacy_archive_brca_manifest.txt`)?

I think if you make changes such that there's a single data download script (https://github.com/greenelab/RNAseq_titration_results/pull/36#issuecomment-857703008), you may end up re-evaluating where `brca_data_urls.txt` is located and have that influence your decision.

It seems tricky to _document_ that unused file, certainly.


> Related, in another simmering PR I add the gdc-client to the docker image. If we aren't using it now, should I add it? Commented out inside Dockerfile? Thank you :)

Sure, that makes sense to me.

**Review by jaclyn-taroni — APPROVED** [review]
Overall this looks good to me. Are the datasets you're getting via the refine.bio client quantile normalized? I would expect you need to skip quantile normalization for this project. Since this is probably just a matter of one option, I don't think I need to take another look.

File: download_TCGA_data.sh [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+# get TCGA array expression using refine.bio client
+# GSE83130 GBM
+# GSE68833 AML
+for accession in GSE83130 GSE68833; do
+  if [ -d $data/$accession ]; then
+    echo refine.bio download for $accession already exists and was not overwritten.
+  else
+    echo Downloading $accession
+    refinebio create-token -s
+    refinebio download-dataset \
```
[the remark this answers is not in the document]
Comment (envest) [reply download_TCGA_data.sh:52]: Added the correct option to not QN 👍

**Pull request merged**